src/Entity/Carrinho.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace App\Entity;
+
+use App\Repository\CarrinhoRepository;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\Mapping as ORM;
+
+#[ORM\Entity(repositoryClass: CarrinhoRepository::class)]
+class Carrinho
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column]
+    private ?int $id = null;
+
+    #[ORM\OneToOne(inversedBy: 'carrinho', cascade: ['persist', 'remove'])]
+    private ?Cliente $cliente_id = null;
+
+    #[ORM\ManyToOne(inversedBy: 'carrinhos')]
+    private ?User $usuario_id = null;
+
+    #[ORM\Column(length: 255)]
+    private ?string $status = null;
+
+    #[ORM\Column]
+    private ?int $valor_total = null;
+
+    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
+    private ?\DateTimeInterface $criado_em = null;
```

Comment:
Sempre que fizer new pode inicializar dentro do construtor

```suggestion
    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private \DateTimeInterface $criadoEm;
```

```php
    public function __construct()
    {
        $this->criadoEm = new DatetimeImmutable();
        $this->items = new ArrayCollection();
    }
    ```